classes/ActionScheduler_Store.php

```diff
@@ -14,6 +14,26 @@
 	/** @var ActionScheduler_Store */
 	private static $store = NULL;
 
+	/**
+	 * Fields that can be stored with actions.
+	 *
+	 * @var array
+	 */
+	protected $action_fields = array(
+		'action_id'            => 1,
+		'hook'                 => 1,
+		'status'               => 1,
+		'scheduled_date_gmt'   => 1,
+		'scheduled_date_local' => 1,
+		'args'                 => 1,
+		'schedule'             => 1,
+		'group_id'             => 1,
```

Comment:
There is no concept of a `group_id` in Action Scheduler, only a group slug/name, which is a string. The group ID exists in custom tables, because it has a group table. But in Action Scheduler itself, a custom taxonomy is used, and only the taxonomy name, not the taxonomy ID, is used to refer to the group.

Reply:
You're absolutely right. The reason I stuck with `group_id` is mainly for the needs of custom tables.

> But in Action Scheduler itself, a custom taxonomy is used, and only the taxonomy name, not the taxonomy ID, is used to refer to the group.

True. However, one gotcha (the good kind?) of the code is that WordPress can accept an ID in place of a slug when using `wp_set_object_terms()`, provided it is an integer. So technically the taxonomy ID (`group_id` here) can still work the same way.

My thought here was that we should model the Custom Tables functionality as the future, and therefore preferred, functionality, and fit the older data store into it. I think I'm just getting ahead of myself here.